src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,16 +612,35 @@ public HttpResponse onSuccess(String authorizationCode) {
         };
     }
     
-    public void updateAuthorizationStrategy(OAuthSession oauth) throws IOException, GeneralSecurityException {
+    public UsernamePasswordAuthenticationToken updateAuthorizationStrategy(Credential credential) throws IOException, GeneralSecurityException {
     	populateDefaults();
-        OpenShiftUserInfo info = getOpenShiftUserInfo(oauth.getCredential(), transport);
-        Set<String> allowedRoles = postSAR(oauth.getCredential(), transport);
+        OpenShiftUserInfo info = getOpenShiftUserInfo(credential, transport);
+        Set<String> allowedRoles = postSAR(credential, transport);
         GrantedAuthority[] authorities = new GrantedAuthority[] {SecurityRealm.AUTHENTICATED_AUTHORITY};
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
```

Comment:
yes but what happens when the initial self-sar gives me view permission and you log me in as "myuser.view" and then on a polling update, you find that i now have admin access?  You can't change my username for the current sesssion(I assume), so you have to update the permissions associated w/ the user i'm currently logged in as (which is myuser.view), at which point you're in the race scenario we had discussed.